aws-logs-logstream/src/main/java/software/amazon/logs/logstream/ReadHandler.java

```diff
@@ -0,0 +1,80 @@
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.SdkClient;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+
+import java.util.Objects;
+
+public class ReadHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final Logger logger) {
+
+        this.logger = logger;
+        final ResourceModel model = request.getDesiredResourceState();
+        logger.log(String.format("LogStreamNamePrefix: %s", model.getLogStreamName()));
+        final String stackId = request.getStackId() == null ? "" : request.getStackId();
+
+        logger.log(String.format("Invoking request for: %s for stack: %s", "AWS-Logs-LogStream::Read", stackId));
+
+        return proxy.initiate("AWS-Logs-LogStream::Read", proxyClient, model, callbackContext)
+                .translateToServiceRequest(Translator::translateToReadRequest)
+                .makeServiceCall((awsRequest, sdkProxyClient) -> {
+                    logger.log(String.format("awsrequest : %s", awsRequest));
+                    return readResource(awsRequest, sdkProxyClient , model, stackId);
+                })
+                .done((awsResponse) -> {
+                    logger.log(String.format("Translator %s", Translator.translateFromReadResponse(awsResponse, model)));
+                    return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                            .status(OperationStatus.SUCCESS)
+                            .resourceModel(Translator.translateFromReadResponse(awsResponse, model))
+                            .build();
+                });
+    }
+
+    private DescribeLogStreamsResponse readResource(
+            final DescribeLogStreamsRequest awsRequest,
+            final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final ResourceModel model,
+            final String stackId) {
+        DescribeLogStreamsResponse describeLogStreamsResponse = null;
+
+        try {
+            describeLogStreamsResponse = proxyClient.injectCredentialsAndInvokeV2(awsRequest, proxyClient.client()::describeLogStreams);
```

Comment:
Use callgraph to make API calls.

Reply:
function is a call in the callgraph. Removed the try-catch statement